mobile/lib/features/trade/channel_creation_flow/channel_configuration_screen.dart

```diff
@@ -0,0 +1,512 @@
+  Amount counterpartyCollateral = Amount.zero();
+
+  double counterpartyLeverage = 1;
+
+  Amount maxOnChainSpending = Amount.zero();
+  Amount maxCounterpartyCollateral = Amount.zero();
+
+  Amount orderMatchingFees = Amount.zero();
+
+  Amount channelFeeReserve = Amount.zero();
+
+  Amount fundingTxFee = Amount.zero();
+
+  final _formKey = GlobalKey<FormState>();
+
+  Usd quantity = Usd.zero();
+
+  Leverage leverage = Leverage(0);
+
+  double liquidationPrice = 0.0;
+
+  bool fundWithWalletEnabled = true;
+
+  Amount maxUsableOnChainBalance = Amount.zero();
+  int maxCounterpartyCollateralSats = 0;
+
+  Amount fundingTxFeeWithBuffer = Amount.zero();
+
+  @override
+  void initState() {
+    super.initState();
+
+    tentenoneConfigChangeNotifier = context.read<TenTenOneConfigChangeNotifier>();
+    var tradeConstraints = tentenoneConfigChangeNotifier.channelInfoService.getTradeConstraints();
+
+    DlcChannelService dlcChannelService =
+        context.read<DlcChannelChangeNotifier>().dlcChannelService;
+
+    quantity = widget.tradeValues.quantity;
+    leverage = widget.tradeValues.leverage;
+    liquidationPrice =
+        widget.tradeValues.liquidationPrice == null ? 0.0 : widget.tradeValues.liquidationPrice!;
+
+    maxCounterpartyCollateral = Amount(tradeConstraints.maxCounterpartyBalanceSats);
+
+    maxOnChainSpending = Amount(tradeConstraints.maxLocalBalanceSats);
+    counterpartyLeverage = tradeConstraints.coordinatorLeverage;
+
+    counterpartyMargin = widget.tradeValues.calculateMargin(Leverage(counterpartyLeverage));
+
+    minMargin = Amount(max(tradeConstraints.minMargin, widget.tradeValues.margin?.sats ?? 0));
+
+    ownTotalCollateral = tradeConstraints.minMargin > widget.tradeValues.margin!.sats
+        ? Amount(tradeConstraints.minMargin)
+        : widget.tradeValues.margin!;
+
+    _collateralController.text = ownTotalCollateral.formatted();
+
+    orderMatchingFees = widget.tradeValues.fee ?? Amount.zero();
+
+    updateCounterpartyCollateral();
+
+    channelFeeReserve = dlcChannelService.getEstimatedChannelFeeReserve();
+
+    fundingTxFee = dlcChannelService.getEstimatedFundingTxFee();
+
+    // We add a buffer because the `fundingTxFee` is just an estimate. This
+    // estimate will undershoot if we end up using more inputs or change
+    // outputs.
+    fundingTxFeeWithBuffer = Amount(fundingTxFee.sats * 2);
+
+    maxUsableOnChainBalance =
+        maxOnChainSpending - orderMatchingFees - fundingTxFeeWithBuffer - channelFeeReserve;
+
+    maxCounterpartyCollateralSats = (maxCounterpartyCollateral.sats * counterpartyLeverage).toInt();
+
+    fundWithWalletEnabled = maxUsableOnChainBalance.sats >= ownTotalCollateral.sats;
+
+    // We add this so that the confirmation slider can be enabled immediately
+    // _if_ the form is already valid. Otherwise we have to wait for the user to
+    // interact with the form.
+    WidgetsBinding.instance.addPostFrameCallback((_) {
```

Comment:
🤔 that looks weird. Is that really the way to go?

Reply:
I've copied this from the old screen 🙈